bin/check-package-metadata.js

```diff
@@ -0,0 +1,149 @@
+ * are registered in all required places and have the expected metadata in their
+ * package.json.
+ */
+'use strict';
+
+const path = require('path');
+const fs = require('fs-extra');
+
+const Project = require('@lerna/project');
+
+async function checkLicenseFiles(packages) {
+  for (const p of packages) {
+    if (!fs.existsSync(path.join(p.location, 'LICENSE'))) {
+      throw new Error(`${p.name} LICENSE file doesn't exist`);
+    }
+  }
+}
+
+async function checkMonorepoFile(packages, rootPath) {
+  const monorepoFile = await fs.readFile('docs/site/MONOREPO.md', 'utf8');
+
+  for (const p of packages) {
+    const packagePath = path.relative(rootPath, p.location);
+
+    const isPackageExist =
+      monorepoFile.includes(p.name) || monorepoFile.includes(packagePath);
+
+    if (!isPackageExist) {
+      throw new Error(
+        `MONOREPO.md doesn't contain mention about the ${p.name}`,
+      );
+    }
+  }
+}
+
+async function checkCodeOwnersFile(packages, rootPath) {
+  const excludes = ['@loopback/sandbox-example'];
+
+  const codeOwnersFile = await fs.readFile('CODEOWNERS', 'utf8');
+
+  for (const p of packages) {
+    if (excludes.includes(p.name)) {
+      continue;
+    }
+
+    const packagePath = path.relative(rootPath, p.location);
+
+    if (!codeOwnersFile.includes(packagePath)) {
+      throw new Error(
+        `CODEOWNERS.md doesn't contain mention about the ${p.name}`,
+      );
+    }
+  }
+}
+
+async function checkPkgsPackageJson(packages, rootPkg) {
+  const pkgs = packages.filter(p => !p.private);
+
+  function getCopyrightOwner(pkg) {
+    return pkg['copyright.owner'] || (pkg.copyright && pkg.copyright.owner);
+  }
+
+  function wrongFieldValueError(packageName, field, original) {
+    throw new Error(
+      `${packageName} package.json has incorrect ${field} field value. It must be ${original}`,
+    );
+  }
+
+  for (const p of pkgs) {
+    const packageName = p.name;
+    const pkg = fs.readJsonSync(path.join(p.location, 'package.json'));
+
+    const isPublicAccess =
+      pkg.publishConfig &&
+      pkg.publishConfig.access &&
+      pkg.publishConfig.access === 'public';
+
+    if (!isPublicAccess) {
+      throw new Error(
+        `${p.name} package.json doesn't contain publicConfig.access equals public`,
+      );
+    }
+
+    if (pkg.author !== rootPkg.author) {
+      wrongFieldValueError(packageName, 'author', rootPkg.author);
+    }
+
+    if (getCopyrightOwner(pkg) !== getCopyrightOwner(rootPkg)) {
+      wrongFieldValueError(
+        packageName,
+        'copyright.owner',
+        getCopyrightOwner(rootPkg),
+      );
+    }
+
+    if (pkg.license !== rootPkg.license) {
+      wrongFieldValueError(packageName, 'license', rootPkg.license);
+    }
+
+    const isCorrectRepositoryUrl =
+      pkg.repository &&
+      pkg.repository.url &&
+      pkg.repository.url.includes(
+        'https://github.com/strongloop/loopback-next',
```

Comment:
I think the repository url should always end with `.git` extension. As far as I understand it, https://github.com/strongloop/loopback-next is pointing to project web page on GitHub, while https://github.com/strongloop/loopback-next.git is pointing to the git repository.

Are there any reasons why we would use https://github.com/strongloop/loopback-next in some places and https://github.com/strongloop/loopback-next.git in others?

I am proposing to use (and enforce) https://github.com/strongloop/loopback-next.git everywhere.

Thoughts?

Reply:
I proposed to remove `.git` for a few reasons:

1. The main usage of repository url is for the npmjs.com to render a link to the github repo. With `.git`, it gets redirected back to https://github.com/strongloop/loopback-next

2. git clone works with both urls (https://github.com/strongloop/loopback-next.git or https://github.com/strongloop/loopback-next).

Reply:
Quoting from https://docs.npmjs.com/files/package.json#repository

> The URL should be a publicly available (perhaps read-only) url that can be handed directly to a VCS program without any modification. It should not be a url to an html project page that you put in your browser. It’s for computers.

The examples in npmjs docs are showing `.git` URLs too.

```
"repository": {
  "type" : "git",
  "url" : "https://github.com/npm/cli.git"
}
```

Regardless of which flavor we decide to use, we should use it consistently in all packages. In the current proposal, some packages are using `loopback-next.git` while others use `loopback-next`.

https://github.com/strongloop/loopback-next/blob/a97266664bb5ec0a379cc131d96e05ea138e3791/packages/cli/package.json#L91-L95

https://github.com/strongloop/loopback-next/blob/a97266664bb5ec0a379cc131d96e05ea138e3791/packages/rest/package.json#L82-L86

Reply:
@spirAde You decide which flavor to use. I agree that consistency is the only purpose.